checks/prometheus_check.py

```diff
@@ -342,18 +424,26 @@ def _submit(self, metric_name, message, send_histograms_buckets=True, custom_tag
         """
         if message.type < len(self.METRIC_TYPES):
             for metric in message.metric:
+                # if hostname is not specified, look at label_to_hostname setting
+                if hostname is None and self.label_to_hostname is not None:
+                    for label in metric.label:
+                        if label.name == self.label_to_hostname:
+                            custom_hostname = label.value
```

Comment:
Are you not afraid of `UnboundLocalError` for **custom_hostname** if label.name == self.label_to_hostname is never True ?

Reply:
There's an `else` bellow setting the variable

Reply:
modified the `else` bellow to `if not custom_hostname`

Reply:
Added a default value above loop